salesagility/SuiteCRM
# Fix #4043 - Workflows with operator 'Equal To' on datetime fields not working properly

Workflows with conditions of type `Start Date` `Equal To` `Now` `+` `5 Minut` never reaches the condition.

## Description

The AOW_WorkFlow build_query_where() method builds the resulting query considering the secconds on datetime fields. So if you create the workflow described on the `Steps to Reproduce` section you obtain a where condition like:

```
meetings.date_start = DATE_ADD(UTC_TIMESTAMP(), INTERVAL - 5 minute)
```

We have modified the build_query_where() method to ignore the secconds when comparing the date fields. So now the obtained query for the previos workflow would be:

```
DATE_FORMAT(meetings.date_start, '%Y-%m-%d %H:%i') = DATE_FORMAT(DATE_ADD(UTC_TIMESTAMP(), INTERVAL - 5 minute), '%Y-%m-%d %H:%i')
```

This is only tested using MySQL. We don't know if this fix works for MSSQL installations.



See issue #4043

## Motivation and Context

We want to send a custom email some time before (or after) of the start date of a meeting.

## How To Test This

1. Create a new workflow with the following parameters
  - WorkFlow Module: Meetings
  - Run: Only In The Scheduler
  - Run On: All Records
  - Repeated Runs: NO
  - Add the following condition
    + Module: Meetings
    + Field: Start Date
    + Operator: Equal To
    + Type: Date
    + Value: Now + 5 Minut
  - Add the following action
    + Select Action: Send Email
    + Email Template: Create new email template with some text.
    + To Email: some_email@example.com
2. Save and activate the workflow.
3. Create a new meeting with a future start_date
4. Note that 5 minutes before the start date of the meeting, the emai is not sent.
5. Apply the changes and see that now the email is received.

## Types of changes


### Final checklist
